File: exercise0/precision.py
"""
FILE: precision.py

Programer: Poruri Sai Rahul rahul.poruri@gmail.com

Date: 22 Jan 2015

Version: Adapted from original

Revision-History:

Comments:
22 Jan 2015 : adapted from the original code, solutions to the various to-do problems
mentioned in the code have been mentioned beneath the original code.

NOTES:
"""
eps = 1.
while (1.+eps != 1.):
	print eps	
	eps = eps/2
print 'machine precision is', eps

File: exercise0/precision.c
/*FILE: precision.c

Programer: Poruri Sai Rahul rahul.poruri@gmail.com

Date: 22 Jan 2015

Version: Original

Revision-History:

Comments:
22 Jan 2015 : code is used to compute machine precision

NOTES:

*/

#include<stdio.h>
#include<stdlib.h>

int
main(void)
{
	float eps;		/*variable to compute machine precision*/
	float i;
	eps = 1;
	i = 1;
	while( i+eps != i)
	{
		eps = eps/2;
		printf("eps %lf\n", eps);
	}
	printf("the machine precision of this system is %e \n",eps);
	return(0);
}
